presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -368,8 +370,18 @@ private boolean isCandidate(Map<ColumnHandle, NullableValue> bindings)
             }
             LookupSymbolResolver inputs = new LookupSymbolResolver(assignments, bindings);
 
+            Object optimized;
+            try {
+                optimized = evaluator.optimize(inputs);
+            }
+            catch (PrestoException e) {
```

Comment:
Are other places which use `evaluator.optimize(inputs)` also vulnerable?

Reply:
Good question. I did a quick search and didn't see anything, but I'm not very familiar with the planner.